### DISCUSSION TITLE
ImGui: docking branch bindings

### DISCUSSION THREAD

**Discussion by Patitotective** [cavariux/nimgl#81] [Ideas]
ImGui bindings are really useful but sometimes I would like to use the docking branch so I was wondering if it was easy to generate them using the [cimgui/docking_inter](https://github.com/cimgui/cimgui/tree/docking_inter) branch.
Thanks :upside_down_face:

**Comment by anddam** [comment]
I tried updating cimgui to 1.89.3 (that has merged the docking branch), I hit a couple gotchas but couldn't get past the most trivial ones like the presence of single digits `ImGuiKey_0` to `ImGuiKey_9` and the presence of `ImGuiKey_End` and `ImGuiKey_END` that marks the end of symbols counting.

There are issues with types too, while reading [this line](https://github.com/cimgui/cimgui/blob/1.89.3/generator/output/typedefs_dict.json#L2) has

    "ImBitArrayForNamedKeys": "ImBitArray<ImGuiKey_NamedKey_COUNT,-ImGuiKey_NamedKey_BEGIN>",

and that value is used in nim type definition (invalid) while it should probably be as [here](https://github.com/cimgui/cimgui/blob/2c8eab86f7ba7ea2b1076a23cb730fed8d8e2b2b/cimgui.h#L1855). In turn this latter symbol has double underscores, invalid in nim identifiers.

So, it'd be nice to have an updated cimgui but I have no idea how to actually translate those.

**Reply by anddam** [reply]
The right side [is C++](https://github.com/cimgui/cimgui/issues/231) and therefore correct as it is.

I will try and see why that generator makes that pop up in `imgui.nim`.

@lmariscal any comment on this?
